openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -517,3 +520,17 @@ def _validate_email(email):
         raise AccountEmailInvalid(
             u"Email '{email}' format is not valid".format(email=email)
         )
+
+
+def set_has_profile_image(username, has_profile_image=True):
```

Comment:
The other "new methods" (get_account_settings, update_account_settings) do very clearly state that the requesting user should match the username (and they enforce this if you pass in the requesting user). Our intention here is that the only code calling set_has_profile_image is the file upload code, and it is being called on behalf of the requesting user. No other code should be calling set_has_profile_image, because it would screw up the synchronization of the the boolean with the actual stored files. By putting the method here in the api file, it implies that it is OK for others to call it.